### PR TITLE
`openingFragment` and `closingFragment` should be JSXOpeningFragment and JSXClosingFragment for JSX fragments

### PR DESCRIPTION
**Pull request by ArtemGovorov**

### CODE REVIEW

**Comment by ArtemGovorov** [comment]
Addressing https://github.com/RReverser/acorn-jsx/pull/83#issuecomment-349726310

**Comment by RReverser** [comment]
Looks like fragments shouldn't have `name` field at all.

**Comment by ArtemGovorov** [comment]
@RReverser Removed the `name` field.

File: inject.js [hunk 1]
```diff
@@ -294,6 +294,7 @@ module.exports = function(acorn) {
       node.attributes.push(this.jsx_parseAttribute());
     node.selfClosing = this.eat(tt.slash);
     this.expect(tt.jsxTagEnd);
+    if (!node.name) delete node.name;
```
Comment (RReverser) [inline inject.js]: `delete` usually hurts performance quite a bit. Could you please read `jsx_parseElementName` into a variable instead and assign to the property only if it's indeed a named element?

Comment (ArtemGovorov) [reply inject.js]: Done.

File: inject.js [hunk 2]
```diff
@@ -289,21 +289,23 @@ module.exports = function(acorn) {
...
     node.attributes = [];
-    node.name = this.jsx_parseElementName();
+    var nodeName = this.jsx_parseElementName();
+    if (nodeName) node.name = nodeName;
     while (this.type !== tt.slash && this.type !== tt.jsxTagEnd)
       node.attributes.push(this.jsx_parseAttribute());
     node.selfClosing = this.eat(tt.slash);
     this.expect(tt.jsxTagEnd);
-    return this.finishNode(node, 'JSXOpeningElement');
+    return this.finishNode(node, node.name ? 'JSXOpeningElement' : 'JSXOpeningFragment');
```
Comment (RReverser) [inline inject.js]: Final nitpick - this can use the same variable now, right?

Comment (ArtemGovorov) [reply inject.js]: Sorry, meant to use it, have forgotten about it. Fixed now.

**Comment by adrianheine** [comment]
Is this in a state were it could be merged? Rich-Harris/buble#62 is waiting for this to be released :)

**Comment by adrianheine** [comment]
Oh, sorry, please take your time :)

**Comment by RReverser** [comment]
Released as 4.1.1 (since technically it's a bug fix, although I really hope no one started depending on the incorrect node type).

**Comment by developit** [comment]
this might have broke rollup/buble (not sure which yet)

**Comment by adrianheine** [comment]
rollup doesn't even use this, and unless you're pulling bublé master you're not seeing this.

**Pull request merged**